test/helpers/quictestlib.c

```diff
@@ -219,7 +219,17 @@ int qtest_create_quic_connection(QUIC_TSERVER *qtserv, SSL *clientssl)
     thread_t t;
 #endif
 
-    if (clientssl == NULL) {
+    /*
+     * Pointless initialisation of t to avoid bogus compiler warnings about
+     * using t uninitialised. Unfortunately there are no well known values
+     * for pthread_t with which to initialise it, so we have to do it like
+     * this instead
+     */
+    memset(&t, 0, sizeof(t));
```

Comment:
Even this might cause warnings on fussy compilers (if t is a pointer type, sizeof(t) has been known to warn).  Wrapping t in a struct:
```c
struct {
    thread_t t;
} t;

memset(&t, 0, sizeof(t));
```
should be safe everywhere.  Use via `t.t`.  Not sure it's worth the pain however.

Another thought which might or might not work:
```c
thread_t t;
static thread_t t_zero;

t = t_zero;
```
but it is a lot cleaner.

Reply:
The second thing should work. You could even make it a static const thread_t t_zero;

Reply:
I went with the second option. Much neater than what I came up with!